eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/TableMapper.java

```diff
@@ -0,0 +1,59 @@
+package io.eventuate.local.mysql.binlog;
+
+import com.github.shyiko.mysql.binlog.event.TableMapEventData;
+import io.eventuate.local.common.SchemaAndTable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TableMapper {
+  private final Map<Long, TableMapEventData> tableMapEventByTableId = new HashMap<>();
+
+  private MySqlBinlogEntryExtractor mySqlBinlogEntryExtractor;
+
+  public TableMapper(MySqlBinlogEntryExtractor mySqlBinlogEntryExtractor) {
+    this.mySqlBinlogEntryExtractor = mySqlBinlogEntryExtractor;
+  }
+
+  public Map<Long, TableMapEventData> getMappings() {
+    return Collections.unmodifiableMap(tableMapEventByTableId);
+  }
+
+  public SchemaAndTable getSchemaAndTable(long tableId) {
+    TableMapEventData tableMapEventData = tableMapEventByTableId.get(tableId);
+
+    return new SchemaAndTable(tableMapEventData.getDatabase(), tableMapEventData.getTable());
+  }
+
+  public void addMapping(TableMapEventData tableMapEventData) {
+    if (tableMapEventByTableId.containsKey(tableMapEventData.getTableId())) {
+      TableMapEventData currentData = tableMapEventByTableId.get(tableMapEventData.getTableId());
+      if (shouldRefreshColumns(currentData, tableMapEventData)) {
+        mySqlBinlogEntryExtractor.refreshColumnOrder();
+      }
+    } else {
+      mySqlBinlogEntryExtractor.refreshColumnOrder();
```

Comment:
rather than calling `mySqlBinlogEntryExtractor.refreshColumnOrder()` addMapping() should return true or false.
This way it does not need to know about `mySqlBinlogEntryExtractor`.

Reply:
ok